spec/prog/vm/github_runner_spec.rb

```diff
@@ -0,0 +1,233 @@
+    end
+
+    it "does not hop to destroy if already in the destroy state" do
+      expect(nx).to receive(:when_destroy_set?).and_yield
+      expect(nx.strand).to receive(:label).and_return("destroy")
+      expect { nx.before_run }.not_to hop("destroy")
+    end
+  end
+
+  describe "#start" do
+    it "register deadline and hop" do
+      expect(nx).to receive(:register_deadline)
+      expect { nx.start }.to hop("wait_vm")
+    end
+  end
+
+  describe "#wait_vm" do
+    it "naps if vm not ready" do
+      expect(vm).to receive(:strand).and_return(Strand.new(label: "prep"))
+      expect { nx.wait_vm }.to nap(5)
+    end
+
+    it "update sshable with IPv4 host and hops" do
+      expect(vm).to receive(:strand).and_return(Strand.new(label: "wait"))
+      expect(vm).to receive(:ephemeral_net4).and_return("1.1.1.1")
+      expect(sshable).to receive(:update).with(host: "1.1.1.1")
+      expect { nx.wait_vm }.to hop("bootstrap_rhizome")
+    end
+
+    it "update sshable with IPv6 host and hops" do
+      expect(vm).to receive(:strand).and_return(Strand.new(label: "wait"))
+      expect(vm).to receive(:ephemeral_net6).and_return(NetAddr::IPv6Net.parse("2001:0db8:85a1::/64"))
+      expect(sshable).to receive(:update).with(host: "2001:db8:85a1::2")
+      expect { nx.wait_vm }.to hop("bootstrap_rhizome")
+    end
+
+    it "update sshable with empty host and hops" do
+      expect(vm).to receive(:strand).and_return(Strand.new(label: "wait"))
+      expect(sshable).to receive(:update).with(host: "")
+      expect { nx.wait_vm }.to hop("bootstrap_rhizome")
+    end
+  end
+
+  describe "#bootstrap_rhizome" do
+    it "buds a bootstrap rhizome process" do
+      expect(nx).to receive(:bud).with(Prog::BootstrapRhizome, {"target_folder" => "common", "subject_id" => vm.id, "user" => "runner"})
+      expect { nx.bootstrap_rhizome }.to hop("wait_bootstrap_rhizome")
+    end
+  end
+
+  describe "#wait_bootstrap_rhizome" do
+    before { expect(nx).to receive(:reap) }
+
+    it "hops to install_actions_runner if there are no sub-programs running" do
+      expect(nx).to receive(:leaf?).and_return true
+
+      expect { nx.wait_bootstrap_rhizome }.to hop("install_actions_runner")
+    end
+
+    it "donates if there are sub-programs running" do
+      expect(nx).to receive(:leaf?).and_return false
+      expect(nx).to receive(:donate).and_call_original
+
+      expect { nx.wait_bootstrap_rhizome }.to nap(0)
+    end
+  end
+
+  describe "#install_actions_runner" do
+    it "downloads and hops to register_runner" do
+      expect(sshable).to receive(:cmd).with(/curl -o actions-runner-linux-x64.*tar.gz/)
+      expect(sshable).to receive(:cmd).with(/echo.*| shasum -a 256 -c/)
+      expect(sshable).to receive(:cmd).with(/tar xzf.*tar.gz/)
+
+      expect { nx.install_actions_runner }.to hop("register_runner")
+    end
+  end
+
+  describe "#register_runner" do
+    it "generates jit runner if not runner id and hops" do
```

Comment:
generates jit runner if runner id is not set and hops

Reply:
Also do we ever create non-jit runner? Maybe specifying it as jit is unnecessary.